File: Kattis/previous/bachetsgame.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

int main() {
  int n, m;
  while (cin >> n >> m) {
    vector<int> A(m);

    for(auto &a:A)cin>>a;

    vector<bool> B(n+1, false);
    B[0] = false;
    for (int i=1; i<=n; i++) {
      bool canWin = false;
      for (int k:A) {
        if (i-k >= 0) {
          if (!B[i-k]) {
            canWin = true;
          }
        }
      }
      B[i] = canWin;
    }
    if (B[n]) {
      cout<<"Stan wins\n";
    }else {
      cout<<"Ollie wins\n";
    }
  }
}
      

File: Kattis/2021Feb04/bachetsgame.py
while(True):
    try:
        i = input().split()
    except EOFError:
        break;

    n = int(i[0])
    m = int(i[1])
    a = list(map(int, i[2:]))

    w = [False]*(n+max(a) +5)
    i = 0
    while i < n+1:
        if not w[i]:
            for j in a:
                w[i+j]=True
            i+=1
        i+=1

    if w[n]:
        print("Stan wins")
    else:
        print("Ollie wins")






